tests/snmp_test.py

```diff
@@ -4,9 +4,102 @@
 from zino.snmp import SNMP
 
 
-@pytest.mark.asyncio
-async def test_snmp(snmpsim, snmp_test_port):
-    device = PollDevice(name="buick.lab.example.org", address="127.0.0.1", port=snmp_test_port)
-    snmp = SNMP(device)
-    response = await snmp.get("SNMPv2-MIB", "sysUpTime", 0)
-    assert response
+class TestSNMPRequests:
+    def get_client(self, port):
+        device = PollDevice(name="buick.lab.example.org", address="127.0.0.1", port=port)
+        return SNMP(device)
```

Comment:
Instead of inheritance, I suggest you use a pytest fixture for this.  This fixture might even depend on the `snmpsim` fixture, and you'll have killed two birds with one stone.

Reply:
good suggestion